src/Compiler/Checking/NameResolution.fs

```diff
@@ -3239,7 +3239,6 @@ let rec ResolvePatternLongIdentPrim sink (ncenv: NameResolver) fullyQualified wa
                 if
                     not newDef
                     && warnOnUpper = WarnOnUpperCase
-                    && id.idText.Length >= 3
```

Comment:
Heh - CI reports an apocalypse :D 

I looked at the original issue - somebody has already done the git blame and seems like the condition has been there forever. Now, looking at the CI errors I have a suspicion this could be just a hack to, well, not get those errors (instead of refactoring code or maybe just renaming the damn variables). 

Moreover, the error that's reported there now is really misleading as seems like there is no pattern matching happening at the erroneous lines.

The good news is that despite the CI apocalypse it's only one file Query.fs :D

Reply:
> Moreover, the error that's reported there now is really misleading as seems like there is no pattern matching happening at the erroneous lines.

Exactly we can update the error message, so it is more clear :). Any suggestion is appreciated

Reply:
I guess it can be something among the same lines but mentioning function parameters. On the other hand, AFAIU this is an orthogonal issue in this context - I'd really prefer just refactoring the code in Query.fs instead.

Reply:
I think for ` function parameters` we could raise a new warning /error with it own message

Reply:
Yep, agreed.

Reply:
@psfinaki  Humm Looking at my last commit to make the CI pass. I think We need to put this behind a preview flag and raised a new warning for the `function parameters` as part of this PR.

Reply:
This is green now  and shows that this `id.idText.Length >= 3` is indeed a hack that has been here forever. I will wait for the team to make a judgment call before investing more time :)

Reply:
Thanks @edgarfgp. Turned to be another rabbit hole but luckily not that deep this time.